src/signal.rs

```diff
@@ -1,13 +1,13 @@
-use tokio::sync::{broadcast, mpsc};
+use tokio::sync::broadcast;
 use tokio_stream::{Stream, StreamExt};
 
 use super::config::ConfigBuilder;
 
 pub type ShutdownTx = broadcast::Sender<()>;
-pub type SignalTx = mpsc::Sender<SignalTo>;
-pub type SignalRx = mpsc::Receiver<SignalTo>;
+pub type SignalTx = broadcast::Sender<SignalTo>;
```

Comment:
I refactored this to use broadcast channels to accommodate the pattern of breaking out of retry loops outside of subcommands.